nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestTreeAnnotationService.java

```diff
@@ -31,4 +51,148 @@
+                createSampleAnnotation(secondDocToAnnotate.getId(), "I am the first annotation of secondFile"));
+
+        // Create first reply on first annotation of first file
+        Annotation firstReply = createAnnotation(
+                createSampleAnnotation(annotationOfFile1.getId(), "I am the first reply of first annotation"));
+
+        // Create second reply
+        Annotation secondReply = createAnnotation(
+                createSampleAnnotation(firstReply.getId(), "I am the second reply of first annotation"));
+
+        // Create third reply
+        Annotation thirdReply = createAnnotation(
+                createSampleAnnotation(secondReply.getId(), "I am the third reply of first annotation"));
+
+        return Map.of( //
+                new IdRef(firstDocToAnnotate.getId()), List.of(annotationOfFile1, firstReply, secondReply, thirdReply), //
+                new IdRef(secondDocToAnnotate.getId()), List.of(annotationOfFile2) //
+        );
+    }
+
+    protected void makeAndVerifyFullTextSearch(String ecmFullText, List<DocumentModel> expectedDocs) {
+        String query = String.format(
+                "SELECT * FROM Document WHERE ecm:fulltext = '%s' AND ecm:mixinType != 'HiddenInNavigation'",
+                ecmFullText);
+
+        DocumentModelList documents = session.query(query);
+        assertEquals(
+                expectedDocs.stream().sorted(Comparator.comparing(DocumentModel::getId)).collect(Collectors.toList()), //
+                documents.stream().sorted(Comparator.comparing(DocumentModel::getId)).collect(Collectors.toList()));
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void checkRelatedTextResource(DocumentRef documentRef, List<Annotation> annotations) {
+        DocumentModel document = session.getDocument(documentRef);
+
+        List<Map<String, String>> resources = (List<Map<String, String>>) document.getPropertyValue(
+                RELATED_TEXT_RESOURCES);
+
+        List<String> relatedTextIds = annotations.stream()
+                                                 .map(a -> String.format(ANNOTATION_RELATED_TEXT_ID, a.getId()))
+                                                 .sorted()
+                                                 .collect(Collectors.toList());
+
+        List<String> relatedTextValues = annotations.stream()
+                                                    .map(Annotation::getText)
+                                                    .sorted()
+                                                    .collect(Collectors.toList());
+
+        assertEquals(relatedTextIds,
+                resources.stream().map(m -> m.get(RELATED_TEXT_ID)).sorted().collect(Collectors.toList()));
+        assertEquals(relatedTextValues,
+                resources.stream().map(m -> m.get(RELATED_TEXT)).sorted().collect(Collectors.toList()));
+    }
+
+    protected Annotation createAnnotation(Annotation annotation) {
```

Comment:
I think it's better to leave transactionnalFeature in the test itself in order to let the developper knows that we wait for. Here we do a lot of nextTransaction (one per annotation) whereas only one call should be enough.

What do you think?

Reply:
after my investigation about the random test failure that i had [here](https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-saouana/141/testReport/)  , i was able to reproduce on my local machine by doing some modification. to fix this problem i should keep `transactionnalFeature` inside the `creationAnnotation` method. we can discuss about it and how i reproduce the problem and why the nextTransaction just after the creation of annotation should fix the isssu .